.devcontainer/Dockerfile

```diff
@@ -1,45 +1,72 @@
-    git-lfs               \
-    cmake                 \
-    ninja-build           \
-    python3-pip           \
-    qtbase5-dev           \
-    libqt5svg5-dev        \
-    graphviz              \
-    doxygen
-
-# Clean ups
-RUN apt-get autoremove -y \
-    && apt-get clean -y   \
-    && rm -rf /var/lib/apt/lists/*
-
-RUN pip3 install \
-  conan          \
-  cmakelint      \
-  gcovr
-
-RUN conan install . \                                                                                                                                                                 SOUR-60 *2 +4 !37 ?14 
--s build_type=Debug \
--of .conan/gcc/ \
--b missing \
--pr:a .conan/gcc/profile
+# Install all dependencies in one step to reduce layers
+RUN apt-get update &&                                                    \
+    apt-get install -y                                                   \
+    build-essential                                                      \
+    wget                                                                 \
+    curl                                                                 \
+    locales                                                              \
+    gcc-12                                                               \
+    g++-12                                                               \
+    mold                                                                 \
+    gdb                                                                  \
+    valgrind                                                             \
+    git                                                                  \
+    git-lfs                                                              \
+    ninja-build                                                          \
+    python3-pip                                                          \
+    qtbase5-dev                                                          \
+    libqt5svg5-dev                                                       \
+    graphviz                                                             \
+    doxygen &&                                                           \
+    update-alternatives --install /usr/bin/gcc gcc /usr/bin/gcc-12 10 && \
+    update-alternatives --install /usr/bin/g++ g++ /usr/bin/g++-12 10 && \
+    apt-get autoremove -y && apt-get clean -y && rm -rf /var/lib/apt/lists/*
 
+# Generate en_US.UTF-8 locale on the system
+RUN locale-gen en_US.UTF-8
+
+# Install CMake in one step
+RUN wget -q https://github.com/Kitware/CMake/releases/download/v3.24.0/cmake-3.24.0-linux-x86_64.tar.gz -O /tmp/cmake.tar.gz && \
+    tar -xzf /tmp/cmake.tar.gz -C /usr/local --strip-components=1 &&                                                            \
+    rm /tmp/cmake.tar.gz
+
+# Remove older version of Clang & LLVM specially 14
+RUN apt-get remove libclang-cpp14 libclang1-14 -y
+
+RUN apt remove --purge libllvm14 -y
+
+RUN ldconfig
```

Comment:
It's fine with me to use clang-15 as compiler at CMakePresets

Reply:
I removed clang as compiler as I decided to go with GCC.
But nevertheless we can return installation of clang compiler but to specify version 15 to avoid incompatibility.

Reply:
I don't think so.